### PR TITLE
[newrelic-logging] Introducing low data mode

### PR DESCRIPTION
**Pull request by fryckbos**
#### Is this a new chart
No.

#### What this PR does / why we need it:
Introduces a low data mode for logs. In this mode only a minimal set of attributes are added to the log data.

#### Special notes for your reviewer:
We need to wait for the release of the new version of the fluentbit image (1.9.3).

#### Checklist
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[mychartname]`)

### CODE REVIEW

File: charts/newrelic-logging/templates/configmap.yaml [hunk 1]
```diff
@@ -27,24 +27,42 @@ data:
...
         Match          kube.*
         # We need the full DNS suffix as Windows only supports resolving names with this suffix
         # See: https://kubernetes.io/docs/setup/production-environment/windows/intro-windows-in-kubernetes/#dns-limitations
         Kube_URL       https://kubernetes.default.svc.cluster.local:443
         K8S-Logging.Exclude ${K8S_LOGGING_EXCLUDE}
+{{- if (include "newrelic.lowDataMode" .) }}
+        Labels         Off
+        Annotations    Off
+
+    [FILTER]
```
Comment (jsubirat) [inline charts/newrelic-logging/templates/configmap.yaml:41]: Small data optimization: actually, given that we're doing this in our pipeline anyway, why don't we apply the lifting by default? The customer won't notice anything in the resulting data format, but we will end up transmitting less data over the wire.

File: charts/newrelic-logging/values.yaml [hunk 2]
```diff
@@ -134,3 +134,8 @@ nrStaging: false
 daemonSet:
   # Annotations to add to the DaemonSet.
   annotations: {}
+
+# When low data mode is enabled only minimal attributes are added to the logs. Kubernetes labels and
+# annotations are not included.
```
Comment (jsubirat) [inline charts/newrelic-logging/values.yaml:139]: Please add a small comment indicating that this also merges `plugin.type`, `plugin.version` and `plugin.source` together into a minified `plugin.source` field.

**Review by jsubirat — COMMENTED** [review]
Really neat work, thanks a lot @fryckbos ! 🎉  I added some comments for you to consider. 🤓

File: charts/newrelic-logging/templates/configmap.yaml [hunk 3]
```diff
@@ -27,24 +27,42 @@ data:
...
+        Operation      lift
+        Nested_under   kubernetes
+{{- end }}
+
+    [FILTER]
+        Name           record_modifier
+        Match          *
+        Record         cluster_name ${CLUSTER_NAME}
+{{- if (include "newrelic.lowDataMode" .) }}
+        Allowlist_key  container_name
```
[the remark this answers is not in the document]
Comment (fryckbos) [reply charts/newrelic-logging/templates/configmap.yaml:53]: It is required, I tested it.

File: charts/newrelic-logging/templates/configmap.yaml (hunk 1 above)
Comment (fryckbos) [reply charts/newrelic-logging/templates/configmap.yaml:41]: For me that's fine, I didn't want to change the default behaviour to limit the blast radius.

Comment (jsubirat) [reply charts/newrelic-logging/templates/configmap.yaml:41]: As you prefer, no strong opinion here.

File: charts/newrelic-logging/templates/configmap.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jsubirat) [reply charts/newrelic-logging/templates/configmap.yaml:53]: So then it is _not_ required, you mean?

[the remark this answers is not in the document]
Comment (fryckbos) [reply charts/newrelic-logging/templates/configmap.yaml:53]: Sorry, I mistyped, it is not required :-)

**Review by tejunior — APPROVED** [review]

**Pull request merged**